.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,153 @@
+name: Setup Poetry
+
+on:
+  workflow_call:
+    inputs:
+      python_version_file:
+        default: '.python-version'
+        type: string
+      working_directory:
+        default: '.'
+        type: string
+      deploy_staging:
+        type: boolean
+      run_typecheck:
+        type: boolean
+      run_datadog_ci:
+        type: boolean
+
+jobs:
+  python_version:
+    uses: ./.github/workflows/python-version.yml
+    with:
+      python_version_file: ${{ inputs.python_version_file }}
+
+  test_and_deploy:
+    name: Test & Deploy
+    needs: python_version
+    runs-on: ubuntu-latest
+    concurrency: test-deploy-${{ github.repository }}
+
+    env:
+      SLS_DEBUG: "*"
+
+    defaults:
+      run:
+        working-directory: ${{ inputs.working_directory }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ needs.python_version.outputs.version }}
```

Comment:
Noticed a `cache: "poetry"` parameter from the old config in https://github.com/GoProperly/template-python-serverless-fastapi/pull/59/files - does that need to be propagated here?

Reply:
And if so, does the poetry install need to be above this config (i.e. does the order matter?)

Reply:
That's a good reminder, thanks. Due to issues with setting up Poetry with `properties` repo, I had skipped caching. But after the fix in `properties` with GoProperly/properties#537 I had not added back caching. I added caching back and the [GH workflow](https://github.com/GoProperly/properties/actions/runs/2951903566) ran through fine.